datafusion/physical-plan/src/unnest.rs

```diff
@@ -515,16 +515,10 @@ fn unnest_list_arrays(
         })
         .collect::<Result<Vec<_>>>()?;
 
-    // If there is only one list column to unnest and it doesn't contain any NULL lists,
-    // we can return the values array directly without any copying.
-    if typed_arrays.len() == 1 && typed_arrays[0].null_count() == 0 {
-        Ok(vec![typed_arrays[0].values().clone()])
```

Comment:
this will return the underlying low level array, which include the row indices filtered out by limit/offset